src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.Tagger.cs

```diff
@@ -3,49 +3,40 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
 using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.Editor.Tagging;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Tagging;
 
-namespace Microsoft.CodeAnalysis.Classification
+namespace Microsoft.CodeAnalysis.Classification;
+
+internal partial class SyntacticClassificationTaggerProvider
 {
-    internal partial class SyntacticClassificationTaggerProvider
+    internal sealed class Tagger : EfficientTagger<IClassificationTag>, IDisposable
     {
-        internal sealed class Tagger : ITagger<IClassificationTag>, IDisposable
-        {
-            private TagComputer? _tagComputer;
-
-            public Tagger(TagComputer tagComputer)
-            {
-                _tagComputer = tagComputer;
-                _tagComputer.TagsChanged += OnTagsChanged;
-            }
-
-            public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
+        private TagComputer? _tagComputer;
 
-            IEnumerable<ITagSpan<IClassificationTag>> ITagger<IClassificationTag>.GetTags(NormalizedSnapshotSpanCollection spans)
-                => GetTags(spans);
```

Comment:
view with whitespace off.  The members above are now provided by the 'EfficientTagger' base class.  The subclass only needs to override AddTags.

Reply:
General question about this. I love that we are doing this, just wondering if we should just rip the bandaid and do it all at once instead of piecemeal like this.

Reply:
I actually like doing it incrementally. It acts as a good way to tell me "how recent was this code touched, and is it likely updated to modern standards.

Reply:
I'm also ok ripping the band-aid off if you want to do that